GERTi/GERTiClient.lua

```diff
@@ -278,12 +412,37 @@ function GERTi.openRoute(destination, outbound)
 			break
 		end
 	end
+	local function cb(r)
+		isOpen = r
+		resultGiven = true
+	end
 	-- if neighbor is local, then open a direct connection
 	if isNeighbor == true then
-		return orController(neighbors[neighborKey]["address"], modem.address, modem.address, neighbors[neighborKey]["address"], neighbors[neighborKey]["port"], neighbors[neighborKey]["port"], outbound)
+		orController(neighbors[neighborKey]["address"], modem.address, modem.address, neighbors[neighborKey]["address"], neighbors[neighborKey]["port"], neighbors[neighborKey]["port"], cb, outbound)
 	else
 		-- if neighbor is not local, then attempt to contact parent to open an indirect connection (i.e. routed through multiple computers)
-		return orController(destination, modem.address, modem.address, neighbors[1]["address"], neighbors[1]["port"], neighbors[1]["port"], outbound)
+		orController(destination, modem.address, modem.address, neighbors[1]["address"], neighbors[1]["port"], neighbors[1]["port"], cb, outbound)
+	end
+	waitWithCancel(10, function () return resultGiven end)
+	return isOpen
+end
+
+-- Yet another low-level function
+function GERTi.resolveAddress(gAddress)
```

Comment:
This function is not needed, as GERTi.openRoute will soon be removed and folded into other functionality to address complexity and space concerns.